articles/django-app-using-postgresql-database/index.md

```diff
@@ -0,0 +1,248 @@
+
+![access psql](/engineering-education/django-app-using-postgresql-database/access_psql.png)
+
+![psql_shell](/engineering-education/django-app-using-postgresql-database/psql_shell.png)
+
+Leave the **Server, Database, Port, Username** options empty. By clicking on `Enter` in the keyboard, you have accepted the default value provided by Shell. Provide the **Password** input with the Password you used during the PostgreSQL Installation on your machine. You will see a similar result in your PSQL shell like the one the above image if correctly implemented.
+
+```bash 
+ CREATE DATABASE helloworld;
+ CREATE USER <yourname> WITH PASSWORD '<password>';
+```
+
+Modifying a few connections parameters for the new database user. 
+
+```bash 
+ALTER ROLE <yourname> SET client_encoding TO 'utf8';
+ALTER ROLE <yourname> SET default_transaction_isolation TO 'read committed';
+ALTER ROLE <yourname> SET timezone TO 'UTC';
+```
+
+Next, we have to grant our new user access to the database we created earlier.
+
+```bash 
+ GRANT ALL PRIVILEGES ON DATABASE helloworld TO <yourname>;
+```
+
+### Step 2: Creating a virtual environment & Installing necessary Python packages
+
+Let's start by creating a folder for our project.
+
+```bash 
+ mkdir myproject 
+ cd myproject
+```
+
+Let's create a virtual environment and activate it.
+
+A virtual environment helps create an isolated python environment which will contain all that packages that our Django project will need. By activating our virtual environment, any Python packages installed will only be available to our Django project.
+
+```bash 
+virtualenv env
+source venv/bin/activate
+```
+
+Let's install `django` and `psycopg2` packages using `pip`. `psycopg2` is a popular PostgreSQL database adapter for python programming language.
+
+```bash
+pip install django psycopg2
+```
+  
+Let's create a Django project.
+
+```bash
+django-admin startproject django_app
+```
+
+Change your current directory into our Django project django_app
+
+```bash
+cd hello_world
+```
+
+Let's create an app in our django_app project.
+
+> A [Django application](https://docs.djangoproject.com/en/3.1/ref/applications/) is a Python package that is specifically intended for use in a Django project. Django apps are reusable in different Django project.
+
+```bash 
+django-admin startapp hello_world
+```
+
+We need to add our `hello_world` app to our `django_app`. To do this, navigate to `django_app/settings.py`.
+
+```python 
+...
+INSTALLED_APPS = [
+    ...
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'hello_world',
+]
+```
+
+### Step 3:  Set up your project to use PostgreSQL Database 
+Navigate to `django_app/settings.py`. In the `DATABASES` section, you'll see the code below.
+
+```python 
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+    }
+}
+```
```

Comment:
Explain that the default database is set to SQLlite and we are going to change it to PostgreSQL